### PR TITLE
net-dns/knot-resolver: update confd/initd with non-privileged user.

### PR DESCRIPTION
**Pull request by PPN-SD**
Use a non-privileged user since the user and group already exists for knot-resolver. See : [https://knot-resolver.readthedocs.io/en/stable/config-no-systemd-privileges.html](https://knot-resolver.readthedocs.io/en/stable/config-no-systemd-privileges.html).
Add "retry" to kill kresd successfully with a big cache.

### CODE REVIEW

**Comment by MatthewGentoo** [comment]
Thanks again! I've [squashed your commits together](https://www.gentoo.org/glep/glep-0066.html#splitting-commits), and [revbumped](https://devmanual.gentoo.org/general-concepts/ebuild-revisions/index.html) the package so that users get the new files.

File: net-dns/knot-resolver/files/kresd.initd [hunk 1]
```diff
@@ -17,7 +26,6 @@ depend() {
     provide dns
 }
 
-start() {
-	checkpath --directory /var/run/kresd
-	default_start
+start_pre() {
+        checkpath -d -m 0750 -o "${KRESD_USER}:${KRESD_GROUP}" ${KRESD_RUNDIR}
```
Comment (Flowdalic) [inline net-dns/knot-resolver/files/kresd.initd:30]: Isn't this applying the anti-pattern mentioned in https://github.com/OpenRC/openrc/blob/master/service-script-guide.md#pid-files-should-be-writable-only-by-root?

**Comment by PPN-SD** [comment]
${KRESD_RUNDIR} is used to store cache essentially. The PID file live in 
/var/run, which is root-owned.

I think it's OK.

Le 2023-09-27 12:42, Florian Schmaus a écrit :

**Comment by Flowdalic** [comment]
Ahh, right. Thanks for the clarification.

**Pull request closed without merging**